### ISSUE TITLE
Fixing opal_timer_linux_find_freq() on x86

### ISSUE THREAD

**Issue by zzzoom** [open-mpi/ompi#3847]
Due to the TSC frequency debacle in #3003 MPI_Wtime got switched back to ```clock_gettime()``` but ```opal_timer_linux_find_freq``` has been left broken. This only affects the one remaining call to ```opal_timer_base_get_freq``` left in ```opal_progress_set_event_poll_rate``` but I'd like to take a shot at fixing it anyway.

Some alternatives I've been thinking about:
1. Measure the frequency between a couple of samples of ```rdtsc``` and ```clock_gettime```. The longer the period the better, but ```opal_timer_linux_find_freq``` doesn't do much so it would take calling ```usleep``` for a few µs or taking the measurements elsewhere during ```opal_init```. There's a small difference between sockets, so context switches + migration would be problematic. Ugh.
2. Take bogomips instead of cpu MHz from ```/proc/cpuinfo```, divide by 2 and hope it keeps working in the future. Linux tries to avoid breaking userspace, so it's probably a safe bet.
3. Make it a configurable parameter and let users/admins worry about it.

Any comments?

**Comment by bosilca** [comment]
According to the discussion on the [MPI Forum \#77](https://github.com/mpi-forum/mpi-issues/issues/77) modern Intel processors have rdtsc ticking at the nominal processor frequency, regardless of power saving and frequency boosting (I paraphrased Jim Dinan).

**Comment by jdinan** [comment]
That's correct -- You can check for invariant TSC to confirm that the processor supports and is configured to provide this behavior.  See: https://software.intel.com/en-us/forums/intel-isa-extensions/topic/280440.  Linux should also set the `constant_tsc` flag in /proc/cpuinfo.
